src/horust/supervisor/process_spawner.rs

```diff
@@ -173,6 +202,43 @@ fn redirect_output(
     Ok(())
 }
 
+fn open_next_chunk(base_path: &Path) -> io::Result<File> {
+    let mut count = 1;
+    let mut path = base_path.to_path_buf();
+
+    while path.is_file() {
+        // let path_str = format!("{}.{count}", base_path,  path.to_string_lossy().to_string() + format!(".{count}").as_ref();
+        let path_str = format!("{}.{count}", base_path.to_string_lossy());
+        // path = Path::new(&path_str).to_path_buf();
+        path = Path::new(&path_str).to_path_buf();
+        count += 1;
+    }
+    debug!("Opening next output: {}", path.display());
+    OpenOptions::new().create(true).append(true).open(&path)
+}
+
+fn chunked_writer(fd: OwnedFd, path: &Path, file_size: u32) -> Result<()> {
+    let path = path.to_path_buf();
+
+    std::thread::spawn(move || {
```

Comment:
I would move the body of the thread to a new function that returns -> io::Error. In that function, you can then use ? to return error results. From inside this block you can then unwrap but only once and map_err(error_printer). The current structure is also ok, but you will need to also handle the unwrap on line 228 with the map_err.

Reply:
Yup, sounds good - switched `chunked_writer()` to just be a function and it is now wrapped by a thread in the main `spawn_process()`.